### PR TITLE
feat: Print start errors in CLI daemon; add `lock-file` package

### PR DESCRIPTION
**Pull request by dmaretskyi**
### Summary
🔒🐛🧹

This pull request adds a new `@dxos/lock-file` package that provides a file-based lock utility, and updates the `@dxos/agent` and `@dxos/client-services` packages to use it. It also improves the `@dxos/agent` package by adding a `Printer` class for printing the error file of the agent process, and by handling the daemon start and stop logic more robustly. It also modifies the `@dxos/devtools` package to run the agent in the foreground or as a daemon depending on the option. It also updates the `release-please-config.json` file to include the new package.

> _`LockFile` package born_
> _A file-based lock utility_
> _Simplifies the code_

### Walkthrough
*  Add a new package `@dxos/lock-file` that provides a file-based lock utility ([link](https://github.com/dxos/dxos/pull/3634/files?diff=unified&w=0#diff-c89da7ad4e85663d87f9031b625bfc5e953a2040208bb7783280fb9b5237521bR1-R30), [link](https://github.com/dxos/dxos/pull/3634/files?diff=unified&w=0#diff-ff091af024a60a2ace7b6e05047bade24a58fa7a5ac2770d12b5da1dd244abcaR1-R65), [link](https://github.com/dxos/dxos/pull/3634/files?diff=unified&w=0#diff-f8ccff0f3839c021ccdb2f014f0206fcaebe7883ba213ad5443cb4374f8f4891R1-R5), [link](https://github.com/dxos/dxos/pull/3634/files?diff=unified&w=0#diff-3a1d0d722df120a528d9d0bcc27aaec38b35598b104870ce183b104228b25e36R1-R70), [link](https://github.com/dxos/dxos/pull/3634/files?diff=unified&w=0#diff-84818a62ffc94127fad615b52621ab59174fd2200db8ba36abc66b44e1a65c23R1-R47), [link](https://github.com/dxos/dxos/pull/3634/files?diff=unified&w=0#diff-0325606061a428c259c1c287ba7d5964643660897c8fcbb61b0c6472b72974b8R1-R33), [link](https://github.com/dxos/dxos/pull/3634/files?diff=unified&w=0#diff-c55d4dcb68c348b2c06d263819702fbce72eeeb35b662956d02049a5a18fcf2bR191-R195)).

### CODE REVIEW

**Comment by mykola-vrmchk** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3634** <a href="https://app.graphite.dev/github/pr/dxos/dxos/3634" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dxos/dxos/3634?utm_source=stack-comment).

File: packages/core/agent/src/forever/forever-daemon.ts [hunk 1]
```diff
@@ -59,35 +60,44 @@ export class ForeverDaemon implements Daemon {
...
-    await waitFor({
-      condition: async () => fs.existsSync(socketFile),
-      timeout: DAEMON_START_TIMEOUT,
-    });
+      await sleep(50);
+
+      const stream = await printFile(errFile);
+      const { path: socketFile } = parseAddress(getUnixSocket(profile));
+      await waitFor({
+        condition: async () => fs.existsSync(socketFile),
```
Comment (dmaretskyi) [inline packages/core/agent/src/forever/forever-daemon.ts]: I dont think that's enough. I'd wanna wait for a successful connection to client-services in the socket and then for the SystemService there to reply with an OK status

File: packages/devtools/cli/src/commands/agent/start.ts [hunk 2]
```diff
@@ -81,10 +84,33 @@ export default class Start extends BaseCommand<typeof Start> {
...
+          this.log(chalk`{red Warning}: '${flags.profile}' is already running (Maybe run 'dx reset')`);
+          return;
+        }
+        try {
+          await daemon.start(this.flags.profile);
+          this.log('Agent started');
+
+          // Check if agent is running.
+          {
+            const services = fromAgent({ profile: this.flags.profile });
```
Comment (dmaretskyi) [inline packages/devtools/cli/src/commands/agent/start.ts]: I'd move this into deamon start

File: packages/core/agent/src/forever/forever-daemon.ts [hunk 3]
```diff
@@ -59,35 +60,44 @@ export class ForeverDaemon implements Daemon {
       mkdirSync(logDir, { recursive: true });
       log('starting...', { profile, logDir });
 
+      const daemonLogFile = path.join(logDir, 'daemon.log');
+      const outFile = path.join(logDir, 'out.log');
+      const errFile = path.join(logDir, 'err.log');
+
+      // Clear err file.
+      fs.unlinkSync(errFile);
```
[the remark this answers is not in the document]
Comment (mykola-vrmchk) [reply packages/core/agent/src/forever/forever-daemon.ts:69]: It is being auto-deleted by forever on each start

File: packages/core/agent/src/printer.ts [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+import { Context } from '@dxos/context';
+
+enum FileState {
+  INIT = 'INIT',
+  READING = 'READING',
+  REACHED_END = 'REACHED_END',
+  DONE = 'DONE',
+}
+
+export class Printer {
```
Comment (mykola-vrmchk) [inline packages/core/agent/src/printer.ts:19]: @dmaretskyi could not make it work

**Pull request merged**